src/diffusers/models/vq_model.py

```diff
@@ -165,17 +167,21 @@ def forward(
             sample (`torch.FloatTensor`): Input sample.
             return_dict (`bool`, *optional*, defaults to `True`):
                 Whether or not to return a [`models.vq_model.VQEncoderOutput`] instead of a plain tuple.
-
+            return_loss (`bool`, *optional*, defaults to `False`):
+                Whether or not to return a commit loss.
         Returns:
             [`~models.vq_model.VQEncoderOutput`] or `tuple`:
                 If return_dict is True, a [`~models.vq_model.VQEncoderOutput`] is returned, otherwise a plain `tuple`
                 is returned.
         """
 
         h = self.encode(sample).latents
-        dec = self.decode(h).sample
+        dec = self.decode(h)
 
         if not return_dict:
-            return (dec,)
-
-        return DecoderOutput(sample=dec)
+            if return_loss:
```

Comment:
maybe we don't need this special `return_loss` flag
I don't think it would break, no?
it is a tuple, we would usually use `out[0]`, if it is a `DecoderOutput`, usually we do `out.sample`; I think just adding the loss to the output should be fine  cc @DN6 to confirm if it's non-breaking

Reply:
Hmm, wouldn't it be possible for some people to do out[-1] for tuple? I think that's the only time it'll break

Reply:
Even if they do that, I think the error message would be fairly easy to digest but I don't think it will be breaking. WDYT? I like the idea of not introducing `return_loss`.

Reply:
Good point. I'll remove return_loss